includes/rest-api/class-sensei-rest-api-questions-controller.php

```diff
@@ -106,13 +106,81 @@ private function serialize_question_as_block( WP_Post $question ) {
 		return serialize_block( $question_block );
 	}
 
+	/**
+	 * Update question block from post content.
+	 *
+	 * @param WP_REST_Request $request
+	 *
+	 * @return WP_Error|WP_Post|WP_REST_Response
+	 */
+	public function update_item( $request ) {
+
+		$block = $this->get_question_block_from_content( $request['content'] );
+
+		if ( $block ) {
+			$request['content'] = '';
+		}
+		parent::update_item( $request );
+
+		if ( $block ) {
+			$this->update_question( $request['id'], $block );
+		}
+
+		// Return the updated question.
+		$response = $this->prepare_item_for_response( get_post( $request['id'] ), $request );
+		return rest_ensure_response( $response );
+
+	}
+
+	/**
+	 * Parse and return question block if it's the first block in the content.
+	 *
+	 * @param string $post_content
+	 *
+	 * @return array|null Question block.
+	 */
+	private function get_question_block_from_content( $post_content ) {
+		if ( ! has_block( 'sensei-lms/quiz-question', $post_content ) ) {
+			return null;
+		}
+		Sensei()->blocks->quiz->initialize_blocks();
+		$block = parse_blocks( trim( $post_content ) )[0] ?? null;
+
+		if ( ! $block || 'sensei-lms/quiz-question' !== $block['blockName'] ) {
+			return null;
+		}
+
+		return $block;
+	}
+
+	/**
+	 * Update question with attributes from the block.
+	 *
+	 * @param int   $id    Question ID.
+	 * @param array $block Question block.
+	 */
+	private function update_question( $id, $block ) {
+		$attrs       = $block['attrs'];
+		$description = serialize_blocks( $block['innerBlocks'] );
+		$question    = array_merge(
+			$attrs,
+			[
+				'description' => $description,
+				'id'          => $id,
+			]
+		);
+
+		$this->save_question( $question );
```

Comment:
It might be nice to handle errors here (`WP_Error` can be returned when title is empty) and somehow fail the update.

Reply:
Updated with some error handling in 4f382ea